### PR TITLE
WIP: Kickoff data download upon Provider save

### PR DESCRIPTION
**Pull request by maskarb**
#1691 

This PR would cause an immediate data ingest when a Source is created.

- Created a receiver on `post_save` signal for Sources. The receiver launches a `check_report_updates` task with the created provider uuid.

- The provider uuid is passed through to the `get_accounts_from_source` function. If this function receives a uuid, it returns the account information for the corresponding provider. If, for whatever reason, that uuid is not found in the list of providers, the function returns an empty list, and no data ingestion occurs. If a provider_uuid is not provided to the function (which is how things currently happen) the function returns the full list of providers like it does today.

### CODE REVIEW

**Comment by adberglund** [comment]
👏

**Comment by adberglund** [comment]
Regarding the concerns: The tasks should get enqueued, so I don't think that's a problem. If we have several large customers, we'll need to scale to support that anyway. 

Second concern is a big one, but theoretically we should have logic that does not duplicate ingestion (download/process) of data. It needs a fix right now, but soon we should have logic that prevents that.

File: koku/api/provider/provider_manager.py [hunk 1]
```diff
@@ -184,6 +185,12 @@ def remove(self, request, customer_remove_context=False):
             raise ProviderManagerError(err_msg)
 
 
+@receiver(post_save, sender=Provider)
+def provider_post_save_callback(sender, instance, created, **kwargs):
+    """Kickoff report download on Provider.save."""
+    check_report_updates.delay(provider_uuid=instance.uuid)
```
Comment (adberglund) [inline koku/api/provider/provider_manager.py]: Do you know if you have information about the request here? If so you could check the request.method and if it's patch *not* call the task?

Comment (adberglund) [reply koku/api/provider/provider_manager.py]: Looks like there is the possibility of a request_finished signal that would likely have info about the request + method.

https://docs.djangoproject.com/en/3.0/ref/signals/#request-finished

**Comment by maskarb** [comment]
Different implementation in [https://github.com/project-koku/koku/pull/1927](https://github.com/project-koku/koku/pull/1927)

**Pull request closed without merging**